### ISSUE TITLE
Enable enforcing coordinator for request

### ISSUE THREAD

**Issue by wprzytula** [scylladb/scylla-rust-driver#1031]
## Motivation

cpp-driver provides an option to explicitly set a node to be coordinator for a specific request. It argues for that option this way:
> In general, this should not be used, but it can be useful in the following situations:
>  - To query node-local tables such as system and virtual tables.
>  - To apply a sequence of schema changes where it makes sense for all the changes to be applied on a single node.

## Solution

Let's expose a setter for explicit coordinator - consider identifying by address or host_id.
If an explicit coordinator is set for a statement, then:
- load balancing policy shall be ignored,
- speculative execution shall be disabled,
- retries shall be only performed on the same node (consider if `RetryNextNode` decisions should be then changed into `RetrySameNode` or into `DontRetry`).

**Comment by wprzytula** [comment]
This should be in line with #738 requirements about enforcing a particular coordinator.

**Comment by Lorak-mmk** [comment]
This can already be done by setting a custom LBP on the query, which yields only 1 element.

**Comment by wprzytula** [comment]
> This can already be done by setting a custom LBP on the query, which yields only 1 element.

Yes, it _can_ be done, but it's fairly inconvenient. The driver's duty is to provide convenient and shorthands to operate with Scylla, and by cpp-driver's documentation (see Motivation) the shorthand for setting an explicit coordinator for a statement is useful.

**Comment by roydahan** [comment]
@muzarski is this one not required for cpp-rust-driver 1.0?

**Comment by muzarski** [comment]
> [@muzarski](https://github.com/muzarski) is this one not required for cpp-rust-driver 1.0?

I just moved it out of the 1.1. I'm not sure if it's going to be prioritized in 1.2 or pushed back to 1.3. Thus, I moved it to 1.x.

**Comment by roydahan** [comment]
If we need to complete cpp-rust-driver work by Q3, it must be either in 1.1 or in 1.2.
If there are other items that are more urgent, we need to understand why.

CC @dkropachev

**Comment by Lorak-mmk** [comment]
> If we need to complete cpp-rust-driver work by Q3, it must be either in 1.1 or in 1.2. If there are other items that are more urgent, we need to understand why.

Why is this implication true? We want to release once per month, so there will be more versions before end of Q3.
